### ISSUE TITLE
make失败

### ISSUE THREAD

**Issue by jireh-H** [aliyun/aliyun-oss-cpp-sdk#128]
Consolidate compiler generated dependencies of target cpp-sdk
[  1%] Building CXX object sdk/CMakeFiles/cpp-sdk.dir/src/external/json/jsoncpp.cpp.o
/home/jireh/aliyun-oss-cpp-sdk-master/aliyun-oss-cpp-sdk-master/sdk/src/external/json/jsoncpp.cpp:1007:34: error: ‘Reader’ is deprecated: Use CharReader and CharReaderBuilder instead [-Werror=deprecated-declarations]
 1007 | Reader::Char Reader::getNextChar() {
      |                                  ^
In file included from /home/jireh/aliyun-oss-cpp-sdk-master/aliyun-oss-cpp-sdk-master/sdk/src/external/json/jsoncpp.cpp:76:
/home/jireh/aliyun-oss-cpp-sdk-master/aliyun-oss-cpp-sdk-master/sdk/src/external/json/json.h:1403:83: note: declared here
 1403 | class JSONCPP_DEPRECATED("Use CharReader and CharReaderBuilder instead") JSON_API Reader {
      |                                                                                   ^~~~~~
/home/jireh/aliyun-oss-cpp-sdk-master/aliyun-oss-cpp-sdk-master/sdk/src/external/json/jsoncpp.cpp:1065:21: error: ‘Reader’ is deprecated: Use CharReader and CharReaderBuilder instead [-Werror=deprecated-declarations]
 1065 | std::vector<Reader::StructuredError> Reader::getStructuredErrors() const {
      |                     ^~~~~~~~~~~~~~~
In file included from /home/jireh/aliyun-oss-cpp-sdk-master/aliyun-oss-cpp-sdk-master/sdk/src/external/json/jsoncpp.cpp:76:
/home/jireh/aliyun-oss-cpp-sdk-master/aliyun-oss-cpp-sdk-master/sdk/src/external/json/json.h:1403:83: note: declared here
 1403 | class JSONCPP_DEPRECATED("Use CharReader and CharReaderBuilder instead") JSON_API Reader {
      |                                                                                   ^~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [sdk/CMakeFiles/cpp-sdk.dir/build.make:384: sdk/CMakeFiles/cpp-sdk.dir/src/external/json/jsoncpp.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:116: sdk/CMakeFiles/cpp-sdk.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

**Comment by zhuzhenxxx** [comment]
大佬解决了吗，我也遇到了

**Comment by zhuzhenxxx** [comment]
cmake -DCMAKE_BUILD_TYPE=debug -DBUILD_STATIC_LIBS=ON -DBUILD_SHARED_LIBS=ON -DARCHIVE_INSTALL_DIR=. -G "Unix Makefiles" ../

**Comment by jinyunshaobing** [comment]
这里主要是因为他们编写的CMakeLists中设置了编译参数-Werror，这会导致所有的警告被视为错误，因此编译会被卡住没法通过。Jsoncpp.cpp中的问题就是这样，Reader被弃用了，代码没更新，就报了弃用的警告导致错误。
要解决也很简单，要么就是帮他们改代码，把报错的地方用CharReader替换掉，要么就是修改CMakeLists.txt文件，搜索-Werror，然后把它删掉或者注释掉

**Comment by danyue333** [comment]
> 这里主要是因为他们编写的CMakeLists中设置了编译参数-Werror，这会导致所有的警告被视为错误，因此编译会被卡住没法通过。Jsoncpp.cpp中的问题就是这样，Reader被弃用了，代码没更新，就报了弃用的警告导致错误。 要解决也很简单，要么就是帮他们改代码，把报错的地方用CharReader替换掉，要么就是修改CMakeLists.txt文件，搜索-Werror，然后把它删掉或者注释掉

感谢，解决了
希望阿里云同学跟进一下

**Comment by Marth-Y** [comment]
> 
感谢，我的openssl 3.0报错也解决了

**Comment by GitBenjamin** [comment]
> 

Bro, how did you solve it? Delete `-Werror`？

**Comment by Marth-Y** [comment]
Hi!&nbsp; ( ˃̶̤́ ꒳ ˂̶̤̀ )
i install openssl3.0 failed.


and then i delete -Werror.and it works.
٩(&nbsp;๑╹&nbsp;ꇴ╹)۶



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2022年11月10日(周四) 下午5:58
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [aliyun/aliyun-oss-cpp-sdk] 安装时 cmake .. 后 make失败 (Issue #128)




   
Bro, how did you solve it? Delete -Werror？

**Comment by GitBenjamin** [comment]
> Hi!&nbsp; ( ˃̶̤́ ꒳ ˂̶̤̀ ) i install openssl3.0 failed. and then i delete -Werror.and it works. ٩(&nbsp;๑╹&nbsp;ꇴ╹)۶
> […](#)
> ---原始邮件--- 发件人: ***@***.***&gt; 发送时间: 2022年11月10日(周四) 下午5:58 收件人: ***@***.***&gt;; 抄送: ***@***.******@***.***&gt;; 主题: Re: [aliyun/aliyun-oss-cpp-sdk] 安装时 cmake .. 后 make失败 (Issue #128) Bro, how did you solve it? Delete -Werror？ — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

Thanks

**Comment by jinyunshaobing** [comment]
> > > 这里主要是因为他们编写的CMakeLists中设置了编译参数-Werror，这会导致所有的警告被视为错误，因此编译会被卡住没法通过。Jsoncpp.cpp中的问题就是这样，Reader被弃用了，代码没更新，就报了弃用的警告导致错误。 要解决也很简单，要么就是帮他们改代码，把报错的地方用CharReader替换掉，要么就是修改CMakeLists.txt文件，搜索-Werror，然后把它删掉或者注释掉
> > 
> > 
> > 感谢，解决了 希望阿里云同学跟进一下
> 
> Bro, how did you solve it? Delete `-Werror`？

Yes, Just delete `-Werror` will solve this problem, `-Werror` will cause all warnings to be treated as errors. So when the `Reader` warning occurs, compile will stop(errors).

**Comment by zhusonger** [comment]
在`list(APPEND SDK_COMPILER_FLAGS "-std=c++11")` 后添加 `list(APPEND SDK_COMPILER_FLAGS "-Wno-deprecated-declarations")`即可

You can append `list(APPEND SDK_COMPILER_FLAGS "-std=c++11")` and then add `list(APPEND SDK_COMPILER_FLAGS "-Wno-deprecated-declarations")`.
